addons/google_gmail/models/google_gmail_mixin.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class GoogleGmailMixin(models.AbstractModel):
+
+    _name = 'google.gmail.mixin'
+    _description = 'Google Gmail Mixin'
+
+    _service_scope = 'https://mail.google.com/'
+
+    use_google_gmail_service = fields.Boolean('Gmail Authentication')
+    google_gmail_authorization_code = fields.Char(string='Authorization Code')
```

Comment:
Considering that mass mailing users can read ir_mail_server shouldn't we add a system group on some of those fields ? Fetchmail server are for admin only.

Reply:
Good idea :) (even if the security issue to bypass it that I reported is still not fixed 🙃)